src/Functions/formatReadableTimeDelta.cpp

```diff
@@ -55,72 +55,88 @@ class FunctionFormatReadableTimeDelta : public IFunction
+                "Number of arguments for function {} doesn't match: passed {}, should be 1, 2 or 3.",
                 getName(), arguments.size());
 
         const IDataType & type = *arguments[0];
 
         if (!isNativeNumber(type))
             throw Exception(ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT, "Cannot format {} as time delta", type.getName());
 
-        if (arguments.size() == 2)
+        if (arguments.size() >= 2)
         {
             const auto * maximum_unit_arg = arguments[1].get();
             if (!isStringOrFixedString(maximum_unit_arg))
                 throw Exception(ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT, "Illegal type {} of argument maximum_unit of function {}",
                                 maximum_unit_arg->getName(), getName());
+
+            if (arguments.size() == 3)
+            {
+                const auto * minimum_unit_arg = arguments[2].get();
+                if (!isStringOrFixedString(minimum_unit_arg))
+                    throw Exception(ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT, "Illegal type {} of argument minimum_unit of function {}",
+                                    minimum_unit_arg->getName(), getName());
+            }
         }
 
         return std::make_shared<DataTypeString>();
     }
 
-    ColumnNumbers getArgumentsThatAreAlwaysConstant() const override { return {1}; }
+    ColumnNumbers getArgumentsThatAreAlwaysConstant() const override { return {1, 2}; }
 
     bool useDefaultImplementationForConstants() const override { return true; }
 
     enum Unit
     {
-        Seconds,
-        Minutes,
-        Hours,
-        Days,
-        Months,
-        Years
+        Nanoseconds = 1,
+        Microseconds = 2,
+        Milliseconds = 3,
+        Seconds = 4,
+        Minutes = 5,
+        Hours = 6,
+        Days = 7,
+        Months = 8,
+        Years = 9
     };
 
     ColumnPtr executeImpl(const ColumnsWithTypeAndName & arguments, const DataTypePtr &, size_t input_rows_count) const override
     {
-        std::string_view maximum_unit_str;
-        if (arguments.size() == 2)
+        std::string_view maximum_unit_str, minimum_unit_str;
+        if (arguments.size() >= 2)
         {
             const ColumnPtr & maximum_unit_column = arguments[1].column;
             const ColumnConst * maximum_unit_const_col = checkAndGetColumnConstStringOrFixedString(maximum_unit_column.get());
             if (maximum_unit_const_col)
                 maximum_unit_str = maximum_unit_const_col->getDataColumn().getDataAt(0).toView();
+
+            if (arguments.size() == 3)
+            {
+                const ColumnPtr & minimum_unit_column = arguments[2].column;
+                const ColumnConst * minimum_unit_const_col = checkAndGetColumnConstStringOrFixedString(minimum_unit_column.get());
+                if (minimum_unit_const_col)
+                    minimum_unit_str = minimum_unit_const_col->getDataColumn().getDataAt(0).toView();
+            }
         }
+        Unit max_unit, min_unit;
```

Comment:
Style is a bit alien here, please fix to something more comon:
```suggestion
        const Unit max_unit = dispatchUnit(maximum_unit_str, Years, "maximum");
        const Unit min_unit = dispatchUnit(minimum_unit_str, Seconds, "minimum");
```
```